lldb/source/Plugins/ABI/LoongArch/ABISysV_loongarch.cpp

```diff
@@ -0,0 +1,664 @@
+                                       heap_data_up->GetBytes() + 0, 8,
+                                       byte_order, error) &&
+          reg_value_a1.GetAsMemoryData(*reg_info_a1,
+                                       heap_data_up->GetBytes() + 8, 8,
+                                       byte_order, error)) {
+        value.SetBytes(heap_data_up.release(), byte_size);
+        return ValueObjectConstResult::Create(
+            thread.GetStackFrameAtIndex(0).get(), value, ConstString(""));
+      }
+    }
+    break;
+  }
+  default:
+    return return_valobj_sp;
+  }
+
+  if (type_flags & eTypeIsInteger) {
+    const bool is_signed = (type_flags & eTypeIsSigned) != 0;
+    if (!SetSizedInteger(value.GetScalar(), raw_value, byte_size, is_signed))
+      return return_valobj_sp;
+  } else if (type_flags & eTypeIsFloat) {
+    if (!SetSizedFloat(value.GetScalar(), raw_value, byte_size))
+      return return_valobj_sp;
+  } else
+    return return_valobj_sp;
+
+  value.SetValueType(Value::ValueType::Scalar);
+  return_valobj_sp = ValueObjectConstResult::Create(
+      thread.GetStackFrameAtIndex(0).get(), value, ConstString(""));
+  return return_valobj_sp;
+}
+
+static ValueObjectSP GetValObjFromFPRegs(Thread &thread,
+                                         const RegisterContextSP &reg_ctx,
+                                         llvm::Triple::ArchType machine,
+                                         uint32_t type_flags,
+                                         uint32_t byte_size) {
+  auto reg_info_fa0 = reg_ctx->GetRegisterInfoByName("f0");
+  bool use_fp_regs = false;
+  ValueObjectSP return_valobj_sp;
+
+  if (byte_size <= 8)
+    use_fp_regs = true;
+
+  if (use_fp_regs) {
+    uint64_t raw_value;
+    Value value;
+    raw_value = reg_ctx->ReadRegisterAsUnsigned(reg_info_fa0, 0);
+    if (!SetSizedFloat(value.GetScalar(), raw_value, byte_size))
+      return return_valobj_sp;
+    value.SetValueType(Value::ValueType::Scalar);
+    return ValueObjectConstResult::Create(thread.GetStackFrameAtIndex(0).get(),
+                                          value, ConstString(""));
+  }
+  // we should never reach this, but if we do, use the integer registers
+  return GetValObjFromIntRegs(thread, reg_ctx, machine, type_flags, byte_size);
+}
+
+ValueObjectSP ABISysV_loongarch::GetReturnValueObjectSimple(
+    Thread &thread, CompilerType &compiler_type) const {
+  ValueObjectSP return_valobj_sp;
+
+  if (!compiler_type)
+    return return_valobj_sp;
+
+  auto reg_ctx = thread.GetRegisterContext();
+  if (!reg_ctx)
+    return return_valobj_sp;
+
+  Value value;
+  value.SetCompilerType(compiler_type);
+
+  const uint32_t type_flags = compiler_type.GetTypeInfo();
+  const size_t byte_size = compiler_type.GetByteSize(&thread).value_or(0);
+  const ArchSpec arch = thread.GetProcess()->GetTarget().GetArchitecture();
+  const llvm::Triple::ArchType machine = arch.GetMachine();
+
+  // Integer return type.
```

Comment:
These comments are redundant, it's clear from the `if (...` what the next block is doing.

Reply:
it's done.